lib/Doctrine/ORM/Tools/SchemaValidator.php

```diff
@@ -298,4 +308,47 @@ public function getUpdateSchemaList(): array
 
         return $schemaTool->getUpdateSchemaSql($allMetadata, true);
     }
+
+    /** @return list<string> containing the found issues */
+    private function validatePropertiesTypes(ClassMetadataInfo $class): array
+    {
+        return array_values(
+            array_filter(
+                array_map(
+                    /** @param FieldMapping $fieldMapping */
+                    static function (array $fieldMapping) use ($class): string|null {
+                        $fieldName    = $fieldMapping['fieldName'];
+                        $propertyType = $class->reflFields[$fieldName]?->getType();
+
+                        // If the field type is not a built-in type, we cannot check it
+                        if (! Type::hasType($fieldMapping['type'])) {
+                            return null;
+                        }
+
+                        // If the property type is not a named type, we cannot check it
+                        if (! ($propertyType instanceof ReflectionNamedType)) {
+                            return null;
+                        }
+
+                        $metadataFieldType = (new DefaultTypedFieldMapper())->getBuiltInType(get_class(Type::getType($fieldMapping['type'])));
+
+                        // If the property type is the same as the metadata field type, we are ok
+                        if ($metadataFieldType === $propertyType->getName()) {
+                            return null;
+                        }
+
+                        return sprintf(
+                            "The field '%s#%s' has the property type '%s' that differs from the metadata field type '%s'.",
+                            $class->name,
+                            $fieldName,
+                            $propertyType->getName(),
+                            $metadataFieldType,
+                        );
+                    },
+                    $class->fieldMappings,
+                ),
+                static fn (?string $issue) => $issue !== null,
```

Comment:
This closure can be omitted, since the default behavior of `array_filter()` is to remove empty-ish values, like `null`, in this case